[FEDE-5203] Upgrade Siren netty version to 4.1.68

GeorgeAp opened the pull request:
Upgrade Siren Netty version from `4.1.48` to `4.1.68`

The netty version `siren-4.1.68-1` would be released once the netty/arrow upgrade works in federate

GeorgeAp committed:
[FEDE-4580] Upgrade Siren netty to 4.1.48.final (#17)

* Prefix the "io.netty.noUnsafe" and
"io.netty.buffer.bytebuf.checkAccessible" properties with the siren
keyword. The netty-common and netty-buffer packages are shaded so that
the classes are relocated to a path prefixed with siren.

* added instructions on how to build netty

* prefix version number with siren keyword

fixed dependency issue with our own artifactory

* disable netty's recycler

* debug for renaud

* bump to 4.1.27.Final

* set siren profile for dev-tools module

* Bump version to siren-4.1.27-2-SNAPSHOT

* If direct memory hasn't been specified by the deployer, set it to the value computed by the static initializer

* Try reflection by default on JDK < 14

* Removed redundant code

* Include JDK 14 in the list of JDKs where reflection is enabled by default

* Release siren-4.1.27-2

* Bump version to siren-4.1.27-3-SNAPSHOT

* Try reflection by default on JDK <= 15

* Release siren-4.1.27-3

* Bump to version siren-4.1.27-4-SNAPSHOT

* [FEDE-3917] netty direct memory counter deprecation (#10)

* Deprecate direct memory counter for Java version >= 9+

* Remove getDirectMemoryCounter()

* Extract io.netty.maxDirectMemory Java system property as a static constant

* Set the release version to siren-4.1.27-4

* Set the release version to siren-4.1.27-5-SNAPSHOT (#11)

* Add io.netty.maxDirectMemory to the list of custom Siren java system property

* Add capability to set the max direct memory from a static field and enforce it optionally

* Fix syntax

* Fix syntax

* Update common/src/main/java/io/netty/util/internal/PlatformDependent.java

Co-authored-by: Renaud <renaud@siren.io>

* Update common/src/main/java/io/netty/util/internal/PlatformDependentCompanion.java

Co-authored-by: Renaud <renaud@siren.io>

* Rename into MaxDirectMemorySetting class

Cleaned README file
Copied methods to compute the default max direct memory

* Simplify handling of the max direct memory

* Fix removal

* Refactoring of maxDirectMemory0() into MaxDirectMemorySetting

* Update common/src/main/java/io/netty/util/internal/MaxDirectMemorySetting.java

Co-authored-by: Stéphane Campinas <stephane.campinas@gmail.com>

* Expand the class description

* Move maxDirectMemory0() to original place

* move description of getting the max direct memory to a clearer spot

* ignore test until it can be run without problem with the rest

* removed ignored test

* release siren-4.1.27-5

* bump netty to siren-4.1.27-6-SNAPSHOT

* Prefix the "io.netty.noUnsafe" and
"io.netty.buffer.bytebuf.checkAccessible" properties with the siren
keyword. The netty-common and netty-buffer packages are shaded so that
the classes are relocated to a path prefixed with siren.

* prefix version number with siren keyword

fixed dependency issue with our own artifactory

* disable netty's recycler

* bump to 4.1.27.Final

* set siren profile for dev-tools module

* [FEDE-3917] netty direct memory counter deprecation (#10)

* Deprecate direct memory counter for Java version >= 9+

* Remove getDirectMemoryCounter()

* Extract io.netty.maxDirectMemory Java system property as a static constant

* Set the release version to siren-4.1.27-4

* Add capability to set the max direct memory from a static field and enforce it optionally

* Refactoring of maxDirectMemory0() into MaxDirectMemorySetting

* Move maxDirectMemory0() to original place

* Prefix the "io.netty.noUnsafe" and
"io.netty.buffer.bytebuf.checkAccessible" properties with the siren
keyword. The netty-common and netty-buffer packages are shaded so that
the classes are relocated to a path prefixed with siren.

* added instructions on how to build netty

* prefix version number with siren keyword

fixed dependency issue with our own artifactory

* disable netty's recycler

* debug for renaud

* bump to 4.1.27.Final

* set siren profile for dev-tools module

* Bump version to siren-4.1.27-2-SNAPSHOT

* If direct memory hasn't been specified by the deployer, set it to the value computed by the static initializer

* Try reflection by default on JDK < 14

* Removed redundant code

* Include JDK 14 in the list of JDKs where reflection is enabled by default

* Release siren-4.1.27-2

* Bump version to siren-4.1.27-3-SNAPSHOT

* Try reflection by default on JDK <= 15

* Release siren-4.1.27-3

* Bump to version siren-4.1.27-4-SNAPSHOT

* [FEDE-3917] netty direct memory counter deprecation (#10)

* Deprecate direct memory counter for Java version >= 9+

* Remove getDirectMemoryCounter()

* Extract io.netty.maxDirectMemory Java system property as a static constant

* Set the release version to siren-4.1.27-4

* Set the release version to siren-4.1.27-5-SNAPSHOT (#11)

* Add io.netty.maxDirectMemory to the list of custom Siren java system property

* Add capability to set the max direct memory from a static field and enforce it optionally

* Fix syntax

* Fix syntax

* Update common/src/main/java/io/netty/util/internal/PlatformDependent.java

Co-authored-by: Renaud <renaud@siren.io>

* Update common/src/main/java/io/netty/util/internal/PlatformDependentCompanion.java

Co-authored-by: Renaud <renaud@siren.io>

* Rename into MaxDirectMemorySetting class

Cleaned README file
Copied methods to compute the default max direct memory

* Simplify handling of the max direct memory

* Refactoring of maxDirectMemory0() into MaxDirectMemorySetting

* Update common/src/main/java/io/netty/util/internal/MaxDirectMemorySetting.java

Co-authored-by: Stéphane Campinas <stephane.campinas@gmail.com>

* Expand the class description

* Move maxDirectMemory0() to original place

* move description of getting the max direct memory to a clearer spot

* ignore test until it can be run without problem with the rest

* removed ignored test

* release siren-4.1.27-5

* bump netty to siren-4.1.27-6-SNAPSHOT

* Prefix the "io.netty.noUnsafe" and
"io.netty.buffer.bytebuf.checkAccessible" properties with the siren
keyword. The netty-common and netty-buffer packages are shaded so that
the classes are relocated to a path prefixed with siren.

* prefix version number with siren keyword

fixed dependency issue with our own artifactory

* disable netty's recycler

* bump to 4.1.27.Final

* set siren profile for dev-tools module

* [FEDE-3917] netty direct memory counter deprecation (#10)

* Deprecate direct memory counter for Java version >= 9+

* Remove getDirectMemoryCounter()

* Extract io.netty.maxDirectMemory Java system property as a static constant

* Set the release version to siren-4.1.27-4

* Add capability to set the max direct memory from a static field and enforce it optionally

* Refactoring of maxDirectMemory0() into MaxDirectMemorySetting

* Move maxDirectMemory0() to original place

* Fix rebase

* Fix MaxDirectMemorySetting and PlatformDependent

* Fix PlatformDependent

* Update readme

* Update as per peer review

* Fix rebase as per peer review

* Fix rebase as per peer review

* Fix as per peer review, remove PlatformDependentCompanion and PlatformDependentMaxDirectMemoryTest, prefix maxCapacity with siren

* Fix readme and checkstyle

* Fix checkstyle

* Update README to make checkBounds false by default

Co-authored-by: Stéphane Campinas <stephane.campinas@gmail.com>
Co-authored-by: Fabio Corneti <info@corneti.com>
Co-authored-by: Martin Anseaume <martin.anseaume@siren.io>
Co-authored-by: Johnny Hujol <itudoben@users.noreply.github.com>
Co-authored-by: Johnny Hujol <johnny.hujol@gmail.com>
Co-authored-by: Renaud <renaud@siren.io>

GeorgeAp committed:
Update readme

GeorgeAp committed:
release siren-4.1.48

GeorgeAp committed:
Bumped version to siren-4.1.48-1-SNAPSHOT (#21)

GeorgeAp committed:
[FEDE-4580] Bumped version to siren-4.1.48-2-SNAPSHOT (#22)

* Bumped version to siren-4.1.48-1-SNAPSHOT

* Bumped version to siren-4.1.48-2-SNAPSHOT

* Bumped version to siren-4.1.48-2-SNAPSHOT

GeorgeAp committed:
Fix rebase

GeorgeAp committed:
Clean up

GeorgeAp committed:
Use japicmp version 0.14.3 instead of version 0.15.3 since maven install with the later fails with
"Failed to execute goal com.github.siom79.japicmp:japicmp-maven-plugin:0.15.3:cmp (default) on project netty-common: Could not find previous version for artifact: io.netty:netty-common"

Signed-off-by: George Apaaboah <george.apaaboah@gmail.com>

scampi commented on common/src/main/java/io/netty/util/internal/MaxDirectMemorySetting.java:
```diff
@@ -0,0 +1,51 @@
+      throw new IllegalArgumentException("Specified max_direct_memory must be greater or equal to 0, got " + m);
+    }
+
+    maxDirectMemory = m;
```
Should we fail if `maxDirectMemory != -1` ? It would mean the property was already set.

GeorgeAp replied:
I am not sure about this, i think the value will be assigned but it might not be used by the system if it is already set

GeorgeAp replied:
Yes, i think we should fail, i have added a check https://github.com/sirensolutions/netty/pull/25/commits/adac7ee842f9bd1100c4a4b8a2025e455006d894, thanks

GeorgeAp replied:
@scampi after cherry-picking this commit https://github.com/sirensolutions/netty/pull/25/commits/c4b48f53dbc3ee2ee821132298e01e7bd317a2c6 the check added here is always failing. From debugging it can be seen that the [MaxDirectMemorySetting.set(maxDirectMemory)](https://github.com/sirensolutions/siren-platform/blob/b45c373d9b1fe222ab3304e11bb61dd62fb38da1/core/src/main/java/io/siren/federate/core/SirenPlatformPlugin.java#L229) is called more than once hence the failure. I have removed the check in this commit https://github.com/sirensolutions/netty/pull/25/commits/113ca33be792366c974f763e11274e13028dfb0d

scampi commented on common/src/main/java/io/netty/util/internal/PlatformDependent.java:
```diff
@@ -132,6 +132,12 @@ public void freeDirectBuffer(ByteBuffer buffer) {
         }
     };
 
+    /**
+     * @deprecated use {@link PlatformDependentCompanion} to set and get the max direct memory instead of this
+     * Java system property.
+     */
+    public static final String JAVA_SYS_PROP_IO_NETTY_MAX_DIRECT_MEMORY = "io.netty.maxDirectMemory";
```
is it time to remove the deprecated setting ?

scampi commented on buffer/pom.xml:
```diff
@@ -24,6 +24,7 @@
   </parent>
 
   <artifactId>netty-buffer</artifactId>
+  <version>siren-4.1.68-1-SNAPSHOT</version>
```
Is the `siren-4.1.68-1` release made once the netty/arrow upgrade works in federate ?

GeorgeAp replied:
Yes, it will be released after the netty/arrow upgrade works in federate

GeorgeAp committed:
Remove deprecated settings and update javadoc.
It is time to remove the JAVA_SYS_PROP_IO_NETTY_MAX_DIRECT_MEMORY settings after being deprecated.

Signed-off-by: George Apaaboah <george.apaaboah@gmail.com>

GeorgeAp committed:
Add a check to verify that max direct memory is not set by the system

Signed-off-by: George Apaaboah <george.apaaboah@gmail.com>

GeorgeAp committed:
Clean up

Signed-off-by: George Apaaboah <george.apaaboah@gmail.com>

manseaume committed:
lazy load direct memory limit in class PlatformDependent

GeorgeAp committed:
Clean up after cherrypick of
https://github.com/sirensolutions/netty/pull/26/commits/ea1bcab8538975bb8dc28d511074543452a3da05

Signed-off-by: George Apaaboah <george.apaaboah@gmail.com>

GeorgeAp committed:
Fix cherry-pick

Signed-off-by: George Apaaboah <george.apaaboah@gmail.com>

GeorgeAp committed:
Remove check to verify that max direct memory is not set by the system

Signed-off-by: George Apaaboah <george.apaaboah@gmail.com>

manseaume reviewed (approved):
LGTM
